### PR TITLE
Update sr-formula and add March7th sheet

### PR DESCRIPTION
**Pull request by nguyentvan7**
## Describe your changes

* Update sr-formula to use premod/final stat calculations
* Update sr-formula to better match the in-game calcs/stats (still a little bit of work needs to be done relating to defense, res, etc)
* Add utils for sheet creation
* Add March7th sheet
* Update UI to display calculated values for character sheet targets
* Set up formula listing for sr-formula
* Add eidolon parsing from datamine
* Add debug calculator

## Issue or discord link

- https://github.com/users/frzyc/projects/2/views/1?pane=issue&itemId=48665814
- https://github.com/users/frzyc/projects/2/views/1?pane=issue&itemId=48671844
- https://github.com/users/frzyc/projects/2/views/1?pane=issue&itemId=48659003

## Testing/validation

Did not validate numbers. But validated site did not crash and all targets are showing as expected for March 7th

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ ] I have commented my code, in hard-to understand areas.
- [ ] I have made corresponding changes to README or wiki.
- [ ] For front-end changes, I have updated the corresponding English translations.
- [ ] Ran `yarn run mini-ci` locally to validate format + lint.
- [ ] If there were format issues, I ran `nx format write` to resolve them automatically.

### CODE REVIEW

File: libs/sr-formula/src/data/char/March7th.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  e4: {
+    dmgInc: eidolon[4][0],
+  },
+  e6: {
+    healMult: eidolon[6][0],
+    healBase: eidolon[6][1],
+  },
+} as const
+
+const { final: _final, char } = self
```
Comment (lantua) [inline libs/sr-formula/src/data/char/March7th.ts:65]: Do we use `_final` here? Seems to be `self.final` below.

File: libs/sr-formula/src/data/char/March7th.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  ...dmg(
+    'talentDmg',
+    type,
+    'atk',
+    dm.talent.dmg,
+    'talent',
+    undefined,
+    'followup',
+    undefined,
+    self.formula.base.add(e4_counter_dmgInc)
```
Comment (lantua) [inline libs/sr-formula/src/data/char/March7th.ts:98]: Should be `selfBuff.formula.base.add(...)`

File: libs/sr-formula/src/data/char/util.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+): TagMapNodeEntries {
+  const abilityLevel = self.char[abilityScalingType]
+  const multi = percent(subscript(abilityLevel, levelScalingMulti))
+  const flat = subscript(abilityLevel, levelScalingFlat)
+  const base = sum(prod(self.final[stat], multi), flat)
+  return customHeal(name, base, arg, ...extra)
+}
+
+export function entriesForChar(data_gen: CharacterDataGen): TagMapNodeEntries {
+  const { final: _final, char } = self
```
Comment (lantua) [inline libs/sr-formula/src/data/char/util.ts:119]: Do we need `_final` here?

File: libs/sr-formula/src/data/common/index.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+const data: TagMapNodeEntries = [
+  ...dmg,
+  ...prep,
+
+  reader.withTag({ src: 'iso', et: 'self' }).reread(reader.src('custom')),
+  reader.withTag({ src: 'agg', et: 'self' }).reread(reader.src('custom')),
+
+  // convert src:char, lightCone, relic to src:agg for accumulation
+  reader.src('agg').add(reader.sum.src('char')),
```
Comment (lantua) [inline libs/sr-formula/src/data/common/index.ts:15]: According to the table in `data/util/tag.ts`, do we also include `src:iso <= sum { src:char }` somewhere?

File: libs/sr-formula/src/util.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      stackInt.add(cmpGE(stackIn.withTag({ member }).max, 1, i + 1))
+    ),
+    members.map((member, i) =>
+      stackOut.withTag({ member }).add(cmpEq(stackInt.max, i + 1, 1))
+    ),
+    // Total Team Stat
+    //
+    // CAUTION:
+    // This formula only works for queries with default `undefined` or `sum` accumulator.
+    // Using this on queries with other accumulators, e.g., `ampMulti` may results in an
```
Comment (lantua) [inline libs/sr-formula/src/util.ts:125]: `ampMulti` example may not be applicable here, though we can leave it as is until a better example for SR comes along

**Review by lantua — COMMENTED** [review]
I only checked the `sr-formula` portion.

File: libs/sr-formula/src/data/common/index.ts (hunk 4 above)
Comment (nguyentvan7) [reply libs/sr-formula/src/data/common/index.ts:15]: Think this is it? https://github.com/frzyc/genshin-optimizer/blob/3badff12590f8e253dc17a72f7fe0548a6ab3444/libs/sr-formula/src/util.ts#L28

**Pull request closed without merging**